pkg/bucketeer/cache/processor/feature_flag_processor.go

```diff
@@ -0,0 +1,231 @@
+	apiClient         api.Client
+	tag               string
+	closeCh           chan struct{}
+	loggers           *log.Loggers
+}
+
+// ProcessorConfig is the config for Processor.
+type FeatureFlagProcessorConfig struct {
+	// Cache
+	Cache cache.Cache
+
+	// PollingInterval is a interval of polling feature flags from the server
+	PollingInterval time.Duration
+
+	// APIClient is the client for Bucketeer service.
+	APIClient api.Client
+
+	// Loggers is the Bucketeer SDK Loggers.
+	Loggers *log.Loggers
+
+	// Tag is the Feature Flag tag
+	// The tag is set when a Feature Flag is created, and can be retrieved from the admin console.
+	Tag string
+}
+
+const (
+	cacheTTL                   = time.Duration(0)
+	featureFlagsIDKey          = "bucketeer_feature_flags_id"
+	featureFlagsRequestedAtKey = "bucketeer_feature_flags_requested_at"
+)
+
+func NewFeatureFlagProcessor(conf *FeatureFlagProcessorConfig) FeatureFlagProcessor {
+	p := &processor{
+		cache:             conf.Cache,
+		featureFlagsCache: cache.NewFeaturesCache(conf.Cache),
+		pollingInterval:   conf.PollingInterval,
+		apiClient:         conf.APIClient,
+		tag:               conf.Tag,
+		closeCh:           make(chan struct{}),
+		loggers:           conf.Loggers,
+	}
+	return p
+}
+
+func (p *processor) Run() {
+	go p.runProcessLoop()
+}
+
+func (p *processor) Close() {
+	p.closeCh <- struct{}{}
+}
+
+func (p *processor) runProcessLoop() {
+	defer func() {
+		p.loggers.Debug("bucketeer/cache: runProcessLoop done")
+	}()
+	ticker := time.NewTicker(p.pollingInterval)
+	defer ticker.Stop()
+	// Update the cache once when starts polling
+	if err := p.updateCache(); err != nil {
+		p.loggers.Errorf("bucketeer/cache: failed to update feature flags cache. Error: %v", err)
+	}
+	for {
+		select {
+		case <-ticker.C:
+			if err := p.updateCache(); err != nil {
+				p.loggers.Errorf("bucketeer/cache: failed to update feature flags cache. Error: %v", err)
+				continue
+			}
+		case <-p.closeCh:
+			return
+		}
+	}
+}
+
+func (p *processor) updateCache() error {
+	ftsID, err := p.getFeatureFlagsID()
+	if err != nil {
+		if err != cache.ErrNotFound {
+			return err
+		}
+		p.loggers.Debug("bucketeer/cache: updateCache featureFlagsID not found")
+	}
+	requestedAt, err := p.getFeatureFlagsRequestedAt()
+	if err != nil {
+		if err != cache.ErrNotFound {
+			return err
+		}
+		p.loggers.Debug("bucketeer/cache: updateCache requestedAt not found")
+	}
+	req := model.NewGetFeatureFlagsRequest(
+		p.tag,
+		ftsID,
+		requestedAt,
+	)
+	// Get the latest cache from the server
+	resp, size, err := p.apiClient.GetFeatureFlags(req)
+	if err != nil {
+		return err
+	}
+	p.loggers.Debugf("bucketeer/cache: GetFeatureFlags response: %v, size: %d", resp, size)
+	// Delete all the local cache and save the new one
+	if resp.ForceUpdate {
+		return p.deleteAllAndSaveLocalCache(resp.FeatureFlagsId, resp.RequestedAt, resp.Features)
+	}
+	// Update only the updated flags
+	return p.updateLocalCache(
```

Comment:
When the `forceUpdate` is false, it only updates the updated and archived flags.